src/azure-cli/azure/cli/__main__.py

```diff
@@ -30,6 +30,19 @@ def cli_main(cli, args):
     return cli.invoke(args)
 
 
+try:
+    if sys.argv[1] == 'cloud' and sys.argv[2] == 'import':
```

Comment:
I don't understand why you have to load the CLOUD_ENDPOINTS here. why it is required when the command is `az cloud import`, why not other command

Reply:
This is mainly for use in air-gapped clouds. Users need to provide an URL and use `az cloud import --endpoint` to first fetch the endpoints before they can use azure cli. When they run other commands, an error will be thrown because no cloud endpoints are set unless they have set an environment variable for the url which is another way to provide the endpoint url.

Reply:
Can we refactor the code to defer the error throw instead of adding the check here